Makefile

```diff
@@ -139,13 +142,61 @@ help-install:
 #
 ###############################################################
 
+### Deployment ###############################################
+#
+release-check:
+	@VERSION=$(shell sh -c "git describe --exact") && \
+	 if test -z "$$VERSION"; then                     \
+		echo VERSION is not set.;                 \
+		echo Please create a git tag;             \
+		exit 100;                                 \
+	else                                              \
+		echo "Releasing version: $$VERSION";      \
+	fi;
+
+release-prep: release-check
+	@VERSION=$(shell sh -c "git describe --exact") && \
+	python setup.py clean;                            \
+	python setup.py sdist;                            \
+	pip install dist/starfish-$$VERSION.tar.gz
```

Comment:
I'm not super keen on pip installing in the user's environment.  I would slightly prefer creating a separate environment to do this in.

Reply:
Ok. I'll re-use `create_venv` to create a `release-venv` (but if a user is trying to perform a release, s/he probably deserves what s/he gets... :smile:)

Reply:
Hmmm.... there's a problem of teaching the other targets how to use the newly created virtualenv, e.g. with:

```
release-venv:
   virtualenv release-venv

release-verify: slow
    ....
```

"slow" isn't going to be aware of `release-venv`

Reply:
So the ugly hack would be `$(MAKE) VIRTUALENV=xyz slow`

A GNU Make specific hack would be:
```
[tt]:/tmp/2> cat Makefile 
def:
	echo $${ABC}

foo : export ABC=123
foo : def
[tt]:/tmp/2> make def
echo ${ABC}

[tt]:/tmp/2> make foo
echo ${ABC}
123
[tt]:/tmp/2> 
```

Reply:
Trying something like this with `SHELL=` since at least `python` and `pytest` would need capturing.